Description: Apply the PDF password length limit to the R6 hash too
Origin: backport, https://gitlab.freedesktop.org/poppler/poppler/-/merge_requests/2398
Bug: https://security-tracker.debian.org/tracker/CVE-2026-107705
Forwarded: not-needed

--- a/poppler/Decrypt.cc
+++ b/poppler/Decrypt.cc
@@ -31,6 +31,7 @@
 //========================================================================
 
 #include <config.h>
+#include <algorithm>
 
 #include <cstdint>
 #include <cstring>
@@ -1770,7 +1771,7 @@
     unsigned char aesKey[16];
     unsigned char BE16byteNumber[16];
 
-    int inputPasswordLength = inputPassword->getLength();
+    const int inputPasswordLength = std::min(inputPassword->getLength(), 127);
     int KLength = 32;
     const int userKeyLength = userKey ? 48 : 0;
     int sequenceLength;
